CriticalMass/LocationManager.swift

```diff
@@ -24,6 +24,13 @@ class LocationManager: NSObject, CLLocationManagerDelegate, LocationProvider {
         }
     }
 
+    private let operationQueue: OperationQueue = {
```

Comment:
Having a separate OperationQueue for the LocationManager is problematic because the Location might be outdated when the request is happening as both operation queues might be out of sync. We should refactor the requesting and getting a location in an Async Operation and integrate it into the RequestManagers Operation queue

Reply:
I think I understand the problem, I need a little time to figure out how to refactor LocationManager code.